client/src/components/mapPopup.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {useRef, useEffect, useState} from 'react';
+import * as styles from './mapPopup.module.scss';
+import Overlay from 'ol/Overlay';
+import {Table} from '@trussworks/react-uswds';
+import {Coordinate} from 'ol/Coordinate';
+import Map from 'ol/Map';
+import {FeatureLike} from 'ol/Feature';
+
+interface IMapPopupProps {
+    map: Map,
+    selectedFeature: FeatureLike;
+    position: Coordinate;
+}
+
+const MapPopup = ({map, selectedFeature, position}: IMapPopupProps) => {
+  const popupContainerElement = useRef<HTMLDivElement>(null);
+  const popupCloserElement = useRef<HTMLAnchorElement>(null);
+  const popupContentElement = useRef<HTMLDivElement>(null);
+  const [currentOverlay, setCurrentOverlay] = useState<Overlay>();
+
```

Comment:
This looks like some witchcraft 👀  What's happening with `useRef<HTMLDivElement>(null)` ? Just for educational purposes.

Reply:
These references are effectively ivars pointing to DOM elements - after initial load, they persist between renderings so you can reference the current html element. More [here](https://reactjs.org/docs/refs-and-the-dom.html)